internal/controllers/s3.go

```diff
@@ -36,114 +35,69 @@ func AwsS3Get(w http.ResponseWriter, r *http.Request) {
 	// then return 200 OK and return.
 	// Note: we want to apply the health check *after* the prefix is stripped.
 	if len(c.HealthCheckPath) > 0 && path == c.HealthCheckPath {
-		w.WriteHeader(http.StatusOK)
-		return
+		res.WriteHeader(http.StatusOK)
+		return nil
 	}
 	// Range header
 	var rangeHeader *string
-	if candidate := r.Header.Get("Range"); !swag.IsZero(candidate) {
+	if candidate := req.Header.Get("Range"); !swag.IsZero(candidate) {
 		rangeHeader = aws.String(candidate)
 	}
 
-	client := service.NewClient(r.Context(), aws.String(config.Config.AwsRegion))
-
-	// Replace path with symlink.json
-	idx := strings.Index(path, "symlink.json")
-	if idx > -1 {
-		replaced, err := replacePathWithSymlink(client, c.S3Bucket, c.S3KeyPrefix+path[:idx+12])
-		if err != nil {
-			code, message := toHTTPError(err)
-			log.Printf("error with replacing path for symlink.json:[%d] %s", code, message)
-			http.Error(w, message, code)
-
-			return
-		}
-
-		path = aws.StringValue(replaced) + path[idx+12:]
-	}
-
 	// Ends with / -> listing or index.html
 	if strings.HasSuffix(path, "/") {
 		if c.DirectoryListing {
-			s3listFiles(w, r, client, c.S3Bucket, c.S3KeyPrefix+path)
-
-			return
+			return s3listFiles(e, c.S3Bucket, c.S3KeyPrefix+path)
 		}
 
 		path += c.IndexDocument
 	}
 	// Get a S3 object
-	obj, err := client.S3get(c.S3Bucket, c.S3KeyPrefix+path, rangeHeader)
+	obj, err := service.S3get(req.Context(), c.S3Bucket, c.S3KeyPrefix+path, rangeHeader)
```

Comment:
does there need to be a 
```
defer obj.Body.Close()
```
here?

Reply:
I don't think that would be needed, the func signature returns data, and not an open handler:
```
func S3get(ctx context.Context, bucket, key string, rangeHeader *string) (*s3.GetObjectOutput, error) {
	req := &s3.GetObjectInput{
		Bucket: aws.String(bucket),
		Key:    aws.String(key),
		Range:  rangeHeader,
	}

	return s3.New(c.Session).GetObjectWithContext(ctx, req)
}
```

The example from the docs doesn't appear to be defering either:
```
svc := s3.New(sess, &aws.Config{
   	DisableRestProtocolURICleaning: aws.Bool(true),
})
out, err := svc.GetObject(&s3.GetObjectInput {
   	Bucket: aws.String("bucketname"),
    	Key: aws.String("//foo//bar//moo"),
})
```
src: https://pkg.go.dev/github.com/aws/aws-sdk-go/service/s3#hdr-Automatic_URI_cleaning